lib/common/marker.js

```diff
@@ -0,0 +1,237 @@
+            sort + '.');
+    }
+
+    return errs;
+}
+
+/*
+ * Returns true if the string "sortField" is also one of the marker's object
+ * field.
+ *
+ * For instance, it returns true if:
+ * - sortField is "create_timestamp" and marker is
+ * "{"uuid": "some-uuid", "create_timestamp": "09-09-2015 08:00:00"}
+ *
+ * It returns false if:
+ * - sortField is "create_timestamp" and marker is
+ * "{"uuid": "some-uuid"}
+ */
+function sortFieldInMarker(sortField, marker) {
+    assert.string(sortField, 'sortField');
+    assert.object(marker, 'marker');
+
+    // If sortField specifies an order with 'field.ASC' or 'field.DESC',
+    // remove the order and just keep the sort field name.
+    sortField = sortField.split('.')[0];
+
+    return Object.keys(marker).some(function (markerField) {
+        return sortField === markerField;
+    });
+}
+
+/*
+ * Returns true if all fields in marker "marker" are also present in the sort
+ * fields represented by "sort", false otherwise
+ *
+ * For instance, it returns true if:
+ * - marker is "{"uuid": "some-uuid", "create_timestamp": "09-09-2015 08:00:00"}
+ * - sort is "create_timestamp"
+ *
+ * It returns false if:
+ * - marker is "{"uuid": "some-uuid", "create_timestamp": "09-09-2015 08:00:00"}
+ * - sort is "brand"
+ */
+function allMarkerFieldsInSort(marker, sort) {
+    assert.object(marker, 'marker must be an object');
+    assert.optionalString(sort, 'sort must be a string or undefined');
+
+    sort = sort || '';
+
+    var markerFields = Object.keys(marker);
+    if (markerFields.length === 0) {
+        // A marker object with no property have all its fields
+        // included in "sort", since it has none.
+        return true;
+    }
+
+    // If sortField specifies an order with 'field.ASC' or 'field.DESC',
+    // remove the order and just keep the sort field name.
+    var sortField = sort.split('.')[0];
+
+    return markerFields.every(function (markerField) {
+        return isStrictTotalOrderField(markerField) ||
+            markerField === sortField;
+        });
+}
+
+/*
+ * Returns true if there's a field that can be used to establish
+ * a strict total order on the vms data set, false otherwise.
+ */
+function strictTotalOrderFieldInMarker(marker) {
+    assert.object(marker, 'marker must be an object');
+
+    return Object.keys(marker).some(isStrictTotalOrderField);
+}
+
+function isStrictTotalOrderField(orderKey) {
+    assert.string(orderKey, 'orderKey must be a string');
+
+    return orderKey === STRICT_TOTAL_ORDER_FIELD;
+}
+
+function strictTotalOrderField() {
+    return STRICT_TOTAL_ORDER_FIELD;
+}
+
+/*
+ * Returns true if the marker object "marker" identifies the VM object "vm",
+ * false otherwise.
+ */
+function markerIdentifiesObject(marker, vm) {
+    assert.object(marker, 'marker must be an object');
+    assert.object(vm, 'vm must be an object');
+
+    return marker[STRICT_TOTAL_ORDER_FIELD] === vm[STRICT_TOTAL_ORDER_FIELD];
```

Comment:
What happens if both `"marker"` and `"vm"` have undefined or null
for the `STRICT_TOTAL_ORDER_FIELD`?